### ISSUE TITLE
Yarn deleting 'node_modules', but then requires it when I command "gatsby develop"

### ISSUE THREAD

**Issue by MonoJackay** [gatsbyjs/gatsby#25318]
## Summary
So based on Gatsby's recommendations it seems they want you using yarn as your package manager. Using gatsby-default-starter as my theme of choice for this project, I wanted to add theme-ui to assist with designing and developing my site! So I use "npm i theme-ui gatsby-plugin-theme-ui" but whenever I attempt to use yarn for any plugins such as "gatsby-plugin-web-font-loader" it removes "node_modules" and after if I try to go into development mode with "gatsby develop" it sends me the error.
I guess what I am asking is should I still be attempting to use yarn as a package manager as recommended or should I just stick with npm as it requires it to be able to run functions? Seems this answer is obvious I just am quite keen to see if there's a way around this or if I'm missing something.

Thanks for all the assistance!

**Comment by liam-brooks** [comment]
just stick you yarn as a default package manager

It's not a good practice to mix package managers

**Comment by pieh** [comment]
> It's not a good practice to mix package managers

This.

It really doesn't matter much if you use npm or yarn - you just need to use one of those consistently. `gatsby new` will install dependencies and on very first run it will ask for preferred package manager - unfortunately we don't have nice way to switch selection - but you can manually edit config file in `<homedir>/.config/gatsby/config.json` find `cli` -> `packageManager` and use `npm` or `yarn` as value depending on what you personally prefer and use, so that gatsby CLI use it and not cause you problems like that

**Comment by MonoJackay** [comment]
> just stick you yarn as a default package manager
> 
> It's not a good practice to mix package managers

Certainly not good practice at all. 
I am just trying to find the best method to using yarn since Gatsby is using it at it's core, plus I will be using one of the Starter Themes which will require yarn. Yet in the Documentation tutorial that is provided by Gatsby, it recommends npm (even when you look into yarn). I know it doesn't matter which one but it a real conflict when you are trying to use yarn and it requires node_modules to be able to run gatsby cli dependencies. Even the same error occurs when solely using Yarn.
I am very confused as to how it is possible to avoid this issue. Can I please understand a little more on avoiding this?

**Comment by MonoJackay** [comment]
> > It's not a good practice to mix package managers
> 
> This.
> 
> It really doesn't matter much if you use npm or yarn - you just need to use one of those consistently. `gatsby new` will install dependencies and on very first run it will ask for preferred package manager - unfortunately we don't have nice way to switch selection - but you can manually edit config file in `<homedir>/.config/gatsby/config.json` find `cli` -> `packageManager` and use `npm` or `yarn` as value depending on what you personally prefer and use, so that gatsby CLI use it and not cause you problems like that

I mentioned to liam-brooks about the documentation but I am still confused as much as before. It doesn't allow me to go further into the project when I want to use `gatsby develop` without having "node_modules" installed. Even when using Yarn 2

**Comment by MonoJackay** [comment]
> > It's not a good practice to mix package managers
> 
> This.
> 
> It really doesn't matter much if you use npm or yarn - you just need to use one of those consistently. `gatsby new` will install dependencies and on very first run it will ask for preferred package manager - unfortunately we don't have nice way to switch selection - but you can manually edit config file in `<homedir>/.config/gatsby/config.json` find `cli` -> `packageManager` and use `npm` or `yarn` as value depending on what you personally prefer and use, so that gatsby CLI use it and not cause you problems like that

And I would change this manually of course, but it doesn't even install .config folder for me. I am surely not doing this correctly? But I am trying incredibly hard to follow all the steps that Gatsby have provided within their documentation. So for someone brand new to this, it is hard to follow what is being done wrong.

**Comment by MonoJackay** [comment]
And another thing I will add to this is it doesn't seem to matter if I use yarn or NPM.
The steps I took to install:
1. Use command `npm install -g gatsby-cli` (into the folder that I will be using for my website). Works as expected.
2. Use command `gatsby new . https://github.com/gatsbyjs/gatsby-starter-default`. Works fine but doesn't install node_modules, not sure if that's the issue.
3. Use `gatsby develop` and this is where I receive the error for 

> There was a problem loading the local develop command. Gatsby may not be installed in your site's "node_modules" directory. Perhaps you need to run "npm install"? You might need to delete your "package-lock.json" as well.

It must have to do with the fact that Yarn 2 is still having problems when attempting to run with Gatsby? Because it seems a lot of people building these sites are running into the same or a similar problem. I will downgrade to 1.22.4 for Yarn and see if that helps.

**Comment by MonoJackay** [comment]
Hey guys! Downgrading to 1.22.4 has made thing run fine! I didn't specify I was testing this out on Yarn 2 and I apologise for that as it seems the assumption was I was using Yarn 1. Sorry for the confusion, next time I will make sure to have all relevant information at my hands so make things more understanding when I seek help and assistance! Thank you!

**Comment by LekoArts** [comment]
Since it seems all questions were answered, I'll close this one :)
